src/components/item/sharing/shortLink/ShortLinksRenderer.tsx

```diff
@@ -0,0 +1,181 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { Box, Dialog, Stack } from '@mui/material';
+
+import {
+  ClientHostManager,
+  Context,
+  ShortLinkPlatform as ShortLinkPlatformConst,
+} from '@graasp/sdk';
+
+import { SHORT_LINK_ID_MAX_LENGTH } from '@/config/constants';
+import { GRAASP_REDIRECTION_HOST } from '@/config/env';
+import { hooks } from '@/config/queryClient';
+import {
+  ShortLinkPlatform,
+  ShortLinkPlatformType,
+  randomAlias,
+  randomString,
+} from '@/utils/shortLink';
+
+import ShortLink from './ShortLink';
+import ShortLinkDialogContent from './ShortLinkDialogContent';
+import ShortLinkSkeleton from './ShortLinkSkeleton';
+
+const { useShortLinksItem, useItemPublishedInformation } = hooks;
+
+type Props = {
+  itemId: string;
+  canAdminShortLink: boolean;
+};
+
+type ShortLinkType = {
+  alias: string;
+  linkId: string;
+  platform: ShortLinkPlatform;
+  url: URL;
+  isShorten: boolean;
+};
+
+const ShortLinksRenderer = ({
+  itemId,
+  canAdminShortLink,
+}: Props): JSX.Element => {
+  const { data: apiLinks, isLoading } = useShortLinksItem(itemId);
+  const { data: publishedEntry } = useItemPublishedInformation({ itemId });
+  const [shortLinks, setShortLinks] = useState<ShortLinkType[]>([]);
+  const [modalOpen, setOpen] = useState(false);
+  const [isNew, setIsNew] = useState(false);
+
+  const [initialAlias, setInitAlias] = useState<string>('');
+  const [initialPlatform, setInitPlatform] = useState<ShortLinkPlatform>(
+    Context.Player,
+  );
+
+  const addNewAlias = useCallback(
+    (
+      platform: ShortLinkPlatform,
+      alias: string,
+      url: URL,
+      isShorten: boolean,
+    ) => {
+      setShortLinks((prevShortLinks) =>
+        [
+          ...prevShortLinks,
+          {
+            linkId: randomString(SHORT_LINK_ID_MAX_LENGTH),
+            alias,
+            platform,
+            url,
+            isShorten,
+          },
+        ].sort((a, b) => a.platform.localeCompare(b.platform)),
+      );
+    },
+    [setShortLinks],
+  );
+
+  const [itemPlatforms, setItemPlatforms] = useState<ShortLinkPlatformType[]>(
+    [],
+  );
+
+  useEffect(() => {
+    setItemPlatforms(
```

Comment:
This should not allows to have many library short links. This effect defines the available platforms for the current item. If the item is not published, the Library is excluded.

This code allows to display the short links more dynamically. If we decide to add analytics in ShortLinkPlatform, it will be automatically displayed in the short links UI.

But maybe I haven't see a potential problem ?